include/picongpu/particles/densityProfiles/FromOpenPMDImpl.hpp

```diff
@@ -166,6 +162,103 @@ namespace picongpu
                 __getTransactionEvent().waitForFinished();
             }
 
+            /** Helper class to convert indices between x-y-z coordinates and a system defined by openPMD API mesh
+             *
+             * The latter is defined by a combination of axisLabels and dataOrder of the given mesh object.
+             * Thus, the transformation is bound to the mesh object given in the constructor, not to openPMD API in
+             * general. However, for brevity in this class it is called "openPMD coordinates"
+             */
+            class IndexConverter
```

Comment:
Can we move this class somewhere into `plugin/misc` or somewhere else in PIConGPU that we can use the class everywhere?

Reply:
Yes, I can do that. Perhaps then it would need some slight generalization in functionality. I didn't really consider it and just programmed what was needed here (and checked there was nothing really reusable as-is). I think this can maybe become a follow-up PR.